views/js/runner/plugins/tools/highlighter/plugin.js

```diff
@@ -64,8 +83,38 @@ define([
             var testConfig      = testData.config || {};
             var pluginShortcuts = (testConfig.shortcuts || {})[this.getName()] || {};
             var hasHighlights   = false;
+            var logger          = loggerFactory('highlighterPlugin');
+
+            /**
+             * @var {Array} highlighters - Highlighters collection
+             * We can run multiple instances of the highlighter plugin on one page:
+             * - one for item-level highlights, which persist for the Test session
+             * - others for stimulus-level highlights, which should persist across multiple sessions (TAO-7617)
+             */
+            var highlighters = [];
+
+            /**
+             * Instantiates new highlighter and adds it to array
+             * @param {Object} options
+             * @param {String} options.className - class applied to highlighted spans
+             * @param {String} options.containerSelector - selector for the unique root DOM node the HL will work on
+             * @param {Array}  options.containersBlackList - list of children which should not receive highlights
+             * @param {String} options.id
+             * @returns {Object} a highlighter instance
+             */
+            function addHighlighter(options) {
```

Comment:
From a structure point of view, would it be possible to extract everything related to the management of the highlighters' collection   (like `addHighlighter` ) to proper function or _may be_ to a separate module, especially for readability. 
For example `addHighlighter`, `getHighlighter`, `getItemHighligher`, `getStimulusHighlighters`, etc.

Reply:
I created a separate `collection` module for this code. Test coverage is better this way too.

Reply:
Great!